include/chemfiles/Property.hpp

```diff
@@ -200,6 +200,47 @@ inline bool operator!=(const Property& lhs, const Property& rhs) {
     return !(lhs == rhs);
 }
 
+/// Various metadata associated with a given kind, accessible at compile time
+template<Property::Kind kind>
+struct property_metadata {
+    /// Type of the property
+    using type = void;
```

Comment:
Mind explaining this line to me? Is it meant to cause an error if a bad template argument is given?

Reply:
Yes, exactly =)